engine/src/main/java/org/terasology/identity/storageServiceClient/StorageServiceWorker.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.identity.storageServiceClient;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.config.Config;
+import org.terasology.config.IdentityStorageServiceConfig;
+import org.terasology.config.SecurityConfig;
+import org.terasology.context.Context;
+import org.terasology.i18n.TranslationSystem;
+import org.terasology.identity.ClientIdentity;
+import org.terasology.identity.PublicIdentityCertificate;
+import org.terasology.logic.console.Console;
+import org.terasology.logic.console.CoreMessageType;
+
+import java.net.URL;
+import java.util.Stack;
+import java.util.Map;
+
+/**
+ * The public interface to this package. Manages a communication session with the storage service server,
+ * can answer for status information queries and can perform asynchronous operations on the storage service.
+ * This class can be in two states, with the default being "logged out".
+ */
+public final class StorageServiceWorker {
+
+    private static final Logger logger = LoggerFactory.getLogger(StorageServiceWorker.class);
+
+    StorageServiceWorkerStatus status = StorageServiceWorkerStatus.LOGGED_OUT;
+    APISession sessionInstance;
+    String loginName;
+    Stack<IdentityBundle> conflictingRemoteIdentities;
```

Comment:
Ok, will use an ArrayDeque. Thanks for the input, didn't know that Stack is now discouraged, I used that since conceptually I just need to access the items at one end so it seemed the simplest data structure. Maybe the queue is also a bit better since conflicts can be solved in the order they were discovered (though this probably isn't very important from a user perspective).